hudi-spark/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -83,6 +83,13 @@ private[hudi] object HoodieSparkSqlWriter {
     val fs = basePath.getFileSystem(sparkContext.hadoopConfiguration)
     var exists = fs.exists(new Path(basePath, HoodieTableMetaClient.METAFOLDER_NAME))
 
+    if (exists && mode == SaveMode.Append) {
+      val existingTableName = new HoodieTableMetaClient(sparkContext.hadoopConfiguration, path.get).getTableConfig.getTableName
+      if (!existingTableName.equals(tblName.get)) {
+        throw new HoodieException(s"hoodie table with name $existingTableName already exist at $basePath")
```

Comment:
actually we could have pushed this check into the guts of `hoodie-client`.. Every write will initialize a HoodieTableMetaClient anyway.. And have it error out from inside, it will save this extra tableMetaClient initialization..  @bhasudha let's file a follow up, if you agree

Reply:
Thanks for the comment @vinothchandar. 

I would suggest the following : 

1. I can use HoodieTableConfig here instead of HoodieTableMetaClient (which seems little  expensive here)

2. I would suggest keeping this check here since the rest of the check based on SaveMode is in this class only and IMHO HoodieTableMetaClient seems like responsible for reading the metadata and the writer should make decisions based on the metadata. Please correct me if I am wrong.

3. I will suggest moving checks at Line number 116 to the beginning of the if section so that we can fail fast and avoid a lot of initialization. Same for the table existence check at 172, it should be moved to the beginning of else section.

Please let me know if it sounds reasonable to you. I can file another Jira for improvement.  

Thanks!

Reply:
@AakashPradeep Thanks for the follow up.. 

For other save modes, e.g OVERWRITE (we first delete the existing dataset), this does not apply anyway.. So I was wondering if we can enforce this at the `HoodieTable.create()` level, which always reads `hoodie.properties` anyway... We can add a `ValidationUtils.checkArgument()` there to simply check the writeConfig table name matches what we read from props file? 

I will let you and @bhasudha take it from here.. :)

Reply:
Thanks !